From 34a27417f47e01ab7a4dc16724bf8041d2d3df84 Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Wed, 11 Feb 2015 16:05:23 -0600
Subject: [PATCH 112/127] bootstats: add printk's to measure boot time in more
 detail

Few distro-tweaks to add printk's to visualize boot time better

Author:    Arjan van de Ven <arjan@linux.intel.com>

Signed-off-by: Miguel Bernal Marin <miguel.bernal.marin@linux.intel.com>
---
 arch/x86/kernel/alternative.c | 4 ++++
 drivers/base/firmware_class.c | 2 ++
 init/main.c                   | 2 +-
 kernel/kmod.c                 | 2 ++
 4 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kernel/alternative.c b/arch/x86/kernel/alternative.c
index 32e14d137416..11fcddcd40db 100644
--- a/arch/x86/kernel/alternative.c
+++ b/arch/x86/kernel/alternative.c
@@ -637,7 +637,9 @@ void __init alternative_instructions(void)
 	 * patching.
 	 */
 
+	printk("clr: Applying alternatives\n");
 	apply_alternatives(__alt_instructions, __alt_instructions_end);
+	printk("clr: Applying alternatives done\n");
 
 #ifdef CONFIG_SMP
 	/* Patch to UP if other cpus not imminent. */
@@ -648,6 +650,8 @@ void __init alternative_instructions(void)
 					    _text, _etext);
 	}
 
+	printk("clr: Applying alternatives smp done\n");
+
 	if (!uniproc_patched || num_possible_cpus() == 1)
 		free_init_pages("SMP alternatives",
 				(unsigned long)__smp_locks,
diff --git a/drivers/base/firmware_class.c b/drivers/base/firmware_class.c
index 19b63d20f5d3..53327f96a5c9 100644
--- a/drivers/base/firmware_class.c
+++ b/drivers/base/firmware_class.c
@@ -1258,6 +1258,8 @@ request_firmware(const struct firmware **firmware_p, const char *name,
 {
 	int ret;
 
+	printk("clr: request_firmware: %s\n", name);
+
 	/* Need to pin this module until return */
 	__module_get(THIS_MODULE);
 	ret = _request_firmware(firmware_p, name, device, NULL, 0,
diff --git a/init/main.c b/init/main.c
index 052481fbe363..ad4f9f137df1 100644
--- a/init/main.c
+++ b/init/main.c
@@ -790,7 +790,7 @@ static int __init_or_module do_one_initcall_debug(initcall_t fn)
 	unsigned long long duration;
 	int ret;
 
-	printk(KERN_DEBUG "calling  %pF @ %i\n", fn, task_pid_nr(current));
+	printk(KERN_DEBUG "calling  %pF @ %i\n", fn, raw_smp_processor_id());
 	calltime = ktime_get();
 	ret = fn();
 	rettime = ktime_get();
diff --git a/kernel/kmod.c b/kernel/kmod.c
index 2f37acde640b..ac6afdee5b50 100644
--- a/kernel/kmod.c
+++ b/kernel/kmod.c
@@ -103,6 +103,8 @@ static int call_modprobe(char *module_name, int wait)
 		NULL
 	};
 
+	printk("clr: call_modprobe: %s   %i \n", module_name, wait);
+
 	char **argv = kmalloc(sizeof(char *[5]), GFP_KERNEL);
 	if (!argv)
 		goto out;
-- 
2.14.1

